CRM/Member/Form/MembershipRenewal.php

```diff
@@ -240,7 +241,11 @@ public function buildQuickForm() {
 
     //CRM-16950
     $taxRates = CRM_Core_PseudoConstant::getTaxRates();
-    $taxRate = CRM_Utils_Array::value($this->allMembershipTypeDetails[$defaults['membership_type_id']]['financial_type_id'], $taxRates);
+    $taxRate = CRM_Utils_Array::value($defaults['financial_type_id'], $taxRates);//CRM-21485
+
+    if (is_array($defaults['membership_type_id'])) { //CRM-21485
+      $defaults['membership_type_id'] = $defaults['membership_type_id'][1];
+    }
```

Comment:
What I meant was we only need this three line of code to fix the issue, i.e, we can get rid of the first commit (https://github.com/civicrm/civicrm-core/pull/11459/commits/d44db0967189c9d8b1970d31b88f7849913cc8d8).

Reply:
Ok I understood. Both of the changes will fix the issue. What is the change you prefer? As I am new to this open source. I am ok with your preference.

Reply:
ok then - can you pls update it with the same?